content/200-concepts/100-components/02-prisma-client/062-computed-fields.mdx

```diff
@@ -10,9 +10,138 @@ Computed fields allow you to derive a new field based on existing data. A common
 
 </TopBlock>
 
-## Deriving a Full Name from a First and Last Name
+## Using a Prisma Client extension
 
-Prisma Client does not yet natively support computed fields, but with a bit of TypeScript magic, you can define a function that accepts a generic as an input then extend that generic to ensure it conforms to a specific structure. Finally, you can return that generic with additional computed fields. Let's see how that might look:
+> Prisma Client extensions are currently in [Preview](/about/prisma/releases#preview).
+
+The following example illustrates how to create a [Prisma Client extension](/concepts/components/prisma-client/client-extensions) that adds a `fullName` computed field at runtime to the `User` model in a Prisma schema.
+
+<TabbedContent tabs={[<FileWithIcon text="Prisma Client extension" icon="code"/> ,<FileWithIcon text="Prisma schema" icon="file"/>]}>
+
+<tab>
+
+```tsx
+import { PrismaClient } from '@prisma/client'
+
+const prisma = new PrismaClient().$extends({
+  result: {
+    user: {
+      fullName: {
+        needs: { firstName: true, lastName: true },
+        compute(user) {
+          return `${user.firstName} ${user.lastName}`
+        },
+      },
+    },
+  },
+})
+
+async function main() {
+  /**
+   * Example query containing the `fullName` computed field in the response
+   */
+  const user = await prisma.user.findMany({
+    take: 5,
+  })
+}
+
+main()
+```
+
+</tab>
+
+<tab>
+
+```prisma copy
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+
+model User {
+  id    Int     @id @default(autoincrement())
+  email String  @unique
+  name  String?
+
+  posts Post[]
+}
+
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  published Boolean @default(true)
+  content   String?
+
+  authorId Int?
+  author   User? @relation(fields: [authorId], references: [id])
+}
+```
+
+</tab>
+
+</TabbedContent>
+
+The query above would return a result with the following structure:
+
+<CodeWithResult>
+
+<cmd></cmd>
+<cmdResult>
```

Comment:
- the  "Show CLI results"-expandable should be used in _direct_ combination with a code block. so here we can either:
  - append it to the previous code block; _or_
  - not use an expandable for the code block (which is what I would prefer here because the return structure is relevant to the content of the page)
- I would also suggest to use `findFirst` so that we only have to show the structure of a single object

Reply:
I intentionally put this outside the Tabbed Content component because I didn't want the output to disappear when the user switched to the schema tab. I don't have any strong opinions but I'll think about it and see what works better